lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -1098,6 +1099,15 @@ void ChargePointImpl::message_callback(const std::string& message) {
             this->send(call_error);
             this->securityEventNotification(ocpp::security_events::INVALIDMESSAGES, message, true);
         }
+    } catch (std::out_of_range e) {
+        // std::out_of_range is used when converting strings to enums
+        EVLOG_error << "JSON exception during handling of message: " << e.what();
```

Comment:
```suggestion
        EVLOG_error << "Out of range exception during handling of message: " << e.what();
```

Reply:
OCPP 1.6J specification 7.44. TriggerMessageStatus states:
"NotImplemented Requested notification cannot be sent because it is either not implemented or unknown."
In the specific failure I was investigating there was a trigger message with an unrecognised requestedMessage. The JSON was correctly formatted. Since the MessageTrigger was not in the enumeration it meets the requirement for a NotImplemented status since the request was unknown.

Reply:
Which one was it? All of them should be supported and in case we get an out of range error it should be because the `requestedMessage` has an invalid value, e.g. is no value defined by the `MessageTrigger` enumeration.